buildstockbatch/base.py

```diff
@@ -747,21 +748,16 @@ def validate_resstock_or_comstock_version(project_file):
 
         buildstock_rb = os.path.join(cfg['buildstock_directory'], 'resources/buildstock.rb')
         if os.path.exists(buildstock_rb):
-            versions = {}
             with open(buildstock_rb, 'r') as f:
-                for line in f:
-                    line = line.strip()
-                    for tool in ['ResStock_Version', 'ComStock_Version', 'BuildStockBatch_Version']:
-                        if line.startswith(tool):
-                            lhs, rhs = line.split('=')
-                            version, _ = rhs.split('#')
-                            versions[tool] = eval(version.strip())
-            BuildStockBatch_Version = versions['BuildStockBatch_Version']
+                versions = dict(
+                    re.findall(r"^\s*(ResStock|ComStock|BuildStockBatch)_Version\s*=\s*'(.+)'", f.read(), re.MULTILINE)
+                )
+            BuildStockBatch_Version = semver.Version.parse(versions['BuildStockBatch'])
             if bsb_version < BuildStockBatch_Version:
```

Comment:
Nah, you can [compare version objects to strings](https://python-semver.readthedocs.io/en/latest/usage/compare-versions.html) and it will automatically convert.